packages/devextreme/js/__internal/ui/button/button.tsx

```diff
@@ -44,9 +71,50 @@ const getCssClasses = (model): string => {
     'dx-button-has-icon': !!icon,
     'dx-button-icon-right': iconPosition !== 'left',
   };
+
   return combineClasses(classesMap);
 };
 
+const omit = <T extends Record<string, unknown>, K extends keyof T>(
+  obj: T,
+  excludedKeys: K[],
+): Omit<T, K> => {
+  const excludedSet = new Set(excludedKeys);
+
+  return Object.keys(obj).reduce((result, key) => {
+    if (!excludedSet.has(key as K)) {
+      result[key] = obj[key];
+    }
+
+    return result;
+  }, {}) as Omit<T, K>;
+};
+
+export type ButtonProps = Record<string, unknown> & BaseWidgetProps & ButtonProperties & {
```

Comment:
hm, so actually this `ButtonProps` type is `Record<string, unknown>`, right? Because it's the most wide from this union. Is it necessary?